hoodie-hadoop-mr/src/main/java/com/uber/hoodie/hadoop/realtime/HoodieRealtimeInputFormat.java

```diff
@@ -135,54 +147,67 @@
         return super.listStatus(job);
     }
 
-
-    private static Configuration addExtraReadColsIfNeeded(Configuration configuration) {
-        String readColNames = configuration.get(ColumnProjectionUtils.READ_COLUMN_NAMES_CONF_STR);
-        String readColIds = configuration.get(ColumnProjectionUtils.READ_COLUMN_IDS_CONF_STR);
-
-        if (!readColNames.contains(HoodieRecord.RECORD_KEY_METADATA_FIELD)) {
-            configuration.set(ColumnProjectionUtils.READ_COLUMN_NAMES_CONF_STR,
-                    readColNames + "," + HoodieRecord.RECORD_KEY_METADATA_FIELD);
-            configuration.set(ColumnProjectionUtils.READ_COLUMN_IDS_CONF_STR,
-                    readColIds + "," + HOODIE_RECORD_KEY_COL_POS);
-            LOG.info(String.format("Adding extra _hoodie_record_key column, to enable log merging cols (%s) ids (%s) ",
-                    configuration.get(ColumnProjectionUtils.READ_COLUMN_NAMES_CONF_STR),
-                    configuration.get(ColumnProjectionUtils.READ_COLUMN_IDS_CONF_STR)));
+    /**
+     * Add a field to the existing fields projected
+     */
+    private static Configuration addProjectionField(Configuration conf, String fieldName,
+        int fieldIndex) {
+        String readColNames = conf.get(ColumnProjectionUtils.READ_COLUMN_NAMES_CONF_STR, "");
+        String readColIds = conf.get(ColumnProjectionUtils.READ_COLUMN_IDS_CONF_STR, "");
+
+        String readColNamesPrefix = readColNames + ",";
+        if (readColNames == null || readColNames.isEmpty()) {
```

Comment:
how can `readColNames` be null or empty here if you are appending a comma above?